<launch>
    <arg name='input_lidar_topic'/>
    <arg name='output_image_topic'/>

    <arg name='height' default='16' />
    <arg name='width' default='1024' />
    <arg name="max_vertical_angle" value="15.0" />
    <arg name="min_vertical_angle" value="-15.0" />
    <arg name="max_horizontal_angle" value="180.0" />
    <arg name="min_horizontal_angle" value="-180.0" />
    <arg name='max_distance' default='100' />
    <arg name='invert_distance' default='true' />
    <arg name='normalize_image' default='true' />
    <arg name='void_value' default='0' />
    <env name="ROSCONSOLE_CONFIG_FILE"
        value="$(find lidar_to_other)/config/rosconsole/debug.conf" />

    <node name='ptcl_to_depth_image' type="pointcloud_to_depth_image" pkg="lidar_to_other"
        output="screen">
        <remap from="~input_lidar_topic" to="$(arg input_lidar_topic)" />
        <remap from="~output_image_topic" to="$(arg output_image_topic)" />
        <param name="height" value="$(arg height)" />
        <param name="width" value="$(arg width)" />
        <param name="max_vertical_angle" value="$(arg max_vertical_angle)" />
        <param name="min_vertical_angle" value="$(arg min_vertical_angle)" />
        <param name="max_horizontal_angle" value="$(arg max_horizontal_angle)" />
        <param name="min_horizontal_angle" value="$(arg min_horizontal_angle)" />
        <param name="max_distance" value="$(arg max_distance)" />
        <param name="invert_distance" value="$(arg invert_distance)" type="bool" />
        <param name="normalize_image" value="$(arg normalize_image)" type="bool" />
        <param name="void_value" value="$(arg void_value)" type="double" />
    </node>
</launch>